website/pages/api-docs/secret/key-management/index.mdx

```diff
@@ -0,0 +1,540 @@
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request DELETE \
+    http://127.0.0.1:8200/v1/keymgmt/kms/example-kms
+```
+
+## Distribute Key to KMS Provider
+
+This endpoint distributes a named key to the KMS provider. The key will be securely delivered
+(i.e., wrapped for protection in transit) following the key import specification of the KMS
+provider. The parameters set cannot be changed after the key has been distributed.
+
+| Method | Path                               |
+| :----- | :--------------------------------- |
+| `PUT`  | `/keymgmt/kms/:name/key/:key_name` |
+
+### Parameters
+
+- `name` `(string: <required>)` – Specifies the name of the KMS provider to distribute the given key
+  to. This is provided as part of the request URL.
+
+- `key_name` `(string: <required>)` – Specifies the name of the key to distribute to the given KMS
+  provider. This is provided as part of the request URL.
+
+- `purpose` `([]string: <required>)` – Specifies the purpose of the key. The purpose defines a set
+  of cryptographic capabilities that the key will have in the KMS provider. A key must have at
+  least one of the supported purposes. Currently, `encrypt`, `decrypt`, `sign`, `verify`, `wrap`,
+  and `unwrap` are supported.
+
+- `protection` `(string: "hsm")` – Specifies the protection of the key. The protection defines
+  where cryptographic operations are performed with the key in the KMS provider. Currently,
+  `software` and `hsm` are supported.
+
+### Sample Payload
+
+```json
+{
+  "protection":"hsm",
+  "purpose":"encrypt,decrypt"
+}
+```
+
+### Sample Request
+
+```shell-session
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request PUT \
+    --data @payload.json \
+    http://127.0.0.1:8200/v1/keymgmt/kms/example-kms/key/example-key
+```
+
+## Read Key in KMS Provider
+
+This endpoint returns information about a key that's been distributed to a KMS provider.
+
+| Method | Path                               |
+| :----- | :--------------------------------- |
+| `GET`  | `/keymgmt/kms/:name/key/:key_name` |
+
+### Parameters
+
+- `name` `(string: <required>)` – Specifies the name of the KMS provider. This is provided as
+  part of the request URL.
+
+- `key_name` `(string: <required>)` – Specifies the name of the key. This is provided as part
+  of the request URL.
+
+### Sample Request
+
+```shell-session
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request GET \
+    http://127.0.0.1:8200/v1/keymgmt/kms/example-kms/key/example-key
+```
+
+### Sample Response
+
+```json
+{
+  "data": {
+    "name": "example-key-<unix_timestamp>",
+    "protection":"hsm",
+    "purpose":"encrypt,decrypt"
+  }
+}
+```
+
+## List Keys in KMS Provider
+
+This endpoint returns a list of all keys that have been distributed to the given KMS
+provider. Many keys can be distributed to a single KMS provider.
+
+| Method | Path                     |
+| :----- | :----------------------- |
+| `LIST` | `/keymgmt/kms/:name/key` |
```

Comment:
nit: I feel like this should be `:name/keys` but that may be a bigger discussion to be had internally